File: src/main/java/com/refinedmods/refinedstorage/api/util/IFilter.java
package com.refinedmods.refinedstorage.api.util;

/**
 * A filter.
 */
public interface IFilter<T> {
    int MODE_WHITELIST = 0;
    int MODE_BLACKLIST = 1;

    /**
     * @return the stack being filtered
     */
    T getStack();

    /**
     * @return the compare flags, see {@link IComparer}
     */
    int getCompare();

    /**
     * @return the mode, whitelist or blacklist
     */
    int getMode();

    /**
     * @return true if this is a mod filter, false otherwise
     */
    boolean isModFilter();
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/util/IFilter.kt
package com.refinedmods.refinedstorage.api.util



/**
 * A filter.
 */
interface IFilter<T> {
    /**
     * @return the stack being filtered
     */
    val stack: T

    /**
     * @return the compare flags, see [IComparer]
     */
    val compare: Int

    /**
     * @return the mode, whitelist or blacklist
     */
    val mode: Int

    /**
     * @return true if this is a mod filter, false otherwise
     */
    val isModFilter: Boolean

    companion object {
        const val MODE_WHITELIST = 0
        const val MODE_BLACKLIST = 1
    }
}